### PR TITLE
feat: Add very simple glossary [INGEST-901]

### LINKED ISSUE #3417: add terminology pop-ups
## Issue

This an issue with:

Platform/SDK documentation (all SDKs)

Sentry Product (all areas)

## Description

Adding key term definitions into the body of text can be clunky and longer to read.
A glossary can often be overwhelming and requires the reader to leave the page they're reading, which can interrupt their comprehension.
With many contributors, there is often no source of truth on the definition of a given term.

### Suggested Solution

Add pop-ups for terms that allow the user to stay on the same page. Also, these pop-ups would all be sourced from one file which would act as the source of truth for definitions.

To be completed: February 28, 2022

### PR DESCRIPTION
**Pull request by untitaker**
Fix #3417

Screenshot 2022-02-28 at 16 31 07

### CODE REVIEW

**Comment by imatwawana** [comment]
@untitaker is there a way for me to test this?

**Comment by untitaker** [comment]
I forgot to post, if you check e.g. the term "DSN" here, you'll see the effects of having the DSN definition in the JS file: https://sentry-docs-git-feat-glossary.sentry.dev/product/cli/configuration/#configuration-values

**Comment by imatwawana** [comment]
Ah okay. Looks good!

**Review by imatwawana — APPROVED** [review]

**Comment by imatwawana** [comment]
I'm thinking to complete this PR, we remove the DNS entry and just push the structure. Then I can circle back and do a separate PR with all the actual terminology added to the JS file. What do you think?

**Comment by imatwawana** [comment]
> @imatwawana I would start with at least a couple definitions instead of rolling everything out all at once, to gather feedback on UI
> 
> like I wouldn't spend effort on finishing _all_ the definitions without getting some preliminary user feedback on the general concept before/meanwhile

Ah okay, gotcha. I'll circle back on this PR and add in a couple more.

File: src/gatsby/plugins/gatsby-remark-terms/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// It's unlikely that there is a way to define glossary in a separate markdown
+// file without blocking all pages processing on processing the glossary file.
+//
+// remark-abbr is Copyright (c) Zeste de Savoir (https://zestedesavoir.com)
+// https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-abbr
+
+const visit = require("unist-util-visit");
+
+const TERMS = {
+  DSN: "Data Source Name: Credentials used in SDK to send data to Sentry.",
```
Comment (imatwawana) [inline src/gatsby/plugins/gatsby-remark-terms/index.js:13]:
```suggestion
  DSN: "Data Source Name: Credentials used in SDK to send data to Sentry.",
  transaction: "A single instance of an activity you want to measure or track, like a page load, page navigation, or asynchronous task.",
  quota: "The monthly number of events — errors, attachments, and transactions — that you pay Sentry to track.",

```

**Pull request merged**